src/hooks/usePersonalData.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useReducer, useContext, Dispatch } from 'react'
```

Comment:
I don't really understand what's going on here so it'd be best to hop on a call and talk about this